scripts/ingest-data.ts

```diff
@@ -0,0 +1,69 @@
+import { RecursiveCharacterTextSplitter } from 'langchain/text_splitter';
+import { OpenAIEmbeddings } from 'langchain/embeddings';
+import { PineconeStore } from 'langchain/vectorstores';
+import { pinecone } from '@/utils/pinecone-client';
+import { RawDocumentLoader } from '@/utils/customDocumentLoader';
+import { PINECONE_INDEX_NAME } from '@/config/pinecone';
+import { STYLE_GUIDES } from '@/config/styleGuides';
+
+const logger = console;
+
+export const run = async () => {
+  try {
+    logger.debug('Loading documents...', STYLE_GUIDES);
+
+    const rawDocs = await Promise.all(
+      STYLE_GUIDES.map(async (styleGuide) => ({
+        tag: styleGuide.tag,
+        files: await Promise.all(
+          styleGuide.files.map((file) => {
+            const loader = new RawDocumentLoader(file);
+
+            return loader.load().then((docs) => docs[0]);
```

Comment:
why only the first element? can we use flatMap?)

Reply:
because it returns only one element

for some reason `load()` and `parse()` should return an array and in the corresponding extended class we return just array with one element